lightly/api/api_workflow_datasets.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from lightly.api.api_workflow_client import ApiWorkflowClient
+
+from typing import List
+
+from lightly.openapi_generated.swagger_client.models.create_entity_response import CreateEntityResponse
+from lightly.openapi_generated.swagger_client.models.dataset_create_request import DatasetCreateRequest
+from lightly.openapi_generated.swagger_client.models.dataset_data import DatasetData
+
+
+class _DatasetsMixin:
+
+    def set_dataset_id(self: ApiWorkflowClient, dataset_name: str):
+        '''Sets the dataset id given the name of the dataset
+
+        Args:
+            dataset_name:
+                The name of the dataset for which the dataset_id
+                should be set as attribute
+
+        Returns:
+
+        Raises: ValueError
+
+        '''
+        current_datasets: List[DatasetData] \
+            = self.datasets_api.get_datasets()
+
+        try:
+            dataset_with_specified_name = next(dataset for dataset in current_datasets if dataset.name == dataset_name)
+            self._dataset_id = dataset_with_specified_name.id
+        except StopIteration:
+            raise ValueError(f"A dataset with the name {dataset_name} does not exist on the web platform. "
+                             f"Please create it first.")
+
+    def create_dataset(self: ApiWorkflowClient, dataset_name: str):
+        '''Creates a dataset on the webplatform
+
+        If a dataset with that name already exists, instead the dataset_id is set.
+        Args:
+            dataset_name:
+                The name of the dataset to be created.
+
+        '''
+        try:
+            self.set_dataset_id(dataset_name)
+        except ValueError:
+            self._create_dataset_without_check_existing(dataset_name=dataset_name)
+
+    def _create_dataset_without_check_existing(self: ApiWorkflowClient, dataset_name: str):
+        ''' Creates a dataset on the webplatform
+
+        No checking if a dataset with such a name already exists is performed.
+        Args:
+            dataset_name:
+                The name of the dataset to be created.
+
+        '''
+        body = DatasetCreateRequest(name=dataset_name)
+        response: CreateEntityResponse = self.datasets_api.create_dataset(body=body)
+        self._dataset_id = response.id
+
+    def create_new_dataset_with_counter(self, dataset_basename: str):
```

Comment:
This looks like a function which is only used for testing. If this is the case I'd prefer to move it to the test file instead of having it here.

Reply:
I could imagine that some users want to just create a dataset on the webapp to do some samplings and do not care about the name of it at all and do not want to need to open the webapp. Thus I would keep it there.

Reply:
I disagree, this is something any user can write with two lines of code and I don't see why we should clutter our code with it. If you insist on keeping it here then I'd suggest to rename it to `create_new_dataset_with_unique_name` (or similar).

Reply:
I disagree that every user can create it with two lines of code, as the function has 10 lines of code. 
I could imagine that the function is needed for a colab tutorial where the user will want to rerun the tutorial without needing to create a new dataset again. 
I agree that the name you proposed is better. 

Another idea I had would be to put the multiple `create_dataset` functions into one and and add a parameter to them describing how to handle it if the dataset already exists:
- `"raise Error"`: If a dataset with that name already exists, raise an Error.
- `"take existing"`: Takes the existing dataset by setting the self._dataset_id
- `"make name unique"`: Adds a counter to the dataset name till it is unique, then creates it.